src/mono/mono/component/event_pipe.c

```diff
@@ -158,12 +158,16 @@ event_pipe_enable (
 
 	if (config_providers) {
 		for (guint32 i = 0; i < providers_len; ++i) {
+			char *provider_name = providers[i].provider_name ? mono_utf16_to_utf8 (providers[i].provider_name, g_utf16_len (providers[i].provider_name), error) : NULL;
```

Comment:
just to be consequent with other runtimes, maybe use ep_rt_utf16_to_utf8_string/ep_rt_utf8_string_free ?